### PR TITLE
Accelerate and fix clang-tidy pipeline

### PR DESCRIPTION
**Pull request by sebwolf-de**
* Runtime: 1:14:50s -> 27:02
* Fix warnings

### CODE REVIEW

File: .ci/run-clang-tidy.py [hunk 1]
```diff
@@ -78,12 +90,22 @@ def make_absolute(f, directory):
...
   """Gets a command line for clang-tidy."""
   start = [clang_tidy_binary]
+  if allow_enabling_alpha_checkers:
+    start.append('-allow-enabling-analyzer-alpha-checkers')
   if header_filter is not None:
     start.append('-header-filter=' + header_filter)
+  if line_filter is not None:
+    start.append('-line-filter=' + line_filter)
+  if use_color is not None:
+    if use_color:
```
Comment (krenzland) [inline .ci/run-clang-tidy.py:105]: Just a hint: None is false

File: .ci/run-clang-tidy.py [hunk 2]
```diff
@@ -236,11 +296,25 @@ def main():
...
-    invocation.append('-p=' + build_path)
-    if args.checks:
-      invocation.append('-checks=' + args.checks)
+    invocation = get_tidy_invocation("", clang_tidy_binary, args.checks,
+                                     None, build_path, args.header_filter,
+                                     args.allow_enabling_alpha_checkers,
+                                     args.extra_arg, args.extra_arg_before,
+                                     args.quiet, args.config_file, args.config,
+                                     args.line_filter, args.use_color,
+                                     args.plugins)
```
Comment (krenzland) [inline .ci/run-clang-tidy.py:316]: This should use named arguments, this list is a bit confusing and invites bugs

**Review by krenzland — CHANGES_REQUESTED** [review]

**Comment by sebwolf-de** [comment]
`run-clang-tidy.py` is taken from https://clang.llvm.org/extra/doxygen/run-clang-tidy_8py_source.html, except for the `file_name_dyn_rup` part. So I'm not going to fix these upstream issues.

**Comment by krenzland** [comment]
Ah, that was not obvious from the diff.

**Review by krenzland — APPROVED** [review]

**Pull request merged**